src/main/connect/connectionManager.ts

```diff
@@ -184,32 +177,34 @@ export class ConnectionManager implements ExtensionComponent, vscode.Disposable
     public areCompleteCredentialsSet(): boolean {
         return !!(this._xrayUrl && this._rtUrl && ((this._username && this._password) || this._accessToken));
     }
-
-    public async verifyCredentials(prompt: boolean): Promise<boolean> {
-        if (!this.areXrayCredentialsSet()) {
-            return false;
+    /**
+     * Test whether the connection is valid and the user has permissions to access Xray and Artifactory.
+     */
+    public async testConnection(prompt: boolean): Promise<TestConnectionStatus> {
+        if (!(await this.testConnectionHealth())) {
+            this.deleteCredentialsFromMemory();
+            return TestConnectionStatus.FailedServerNotFound;
         }
         if (
-            !(await ConnectionUtils.checkXrayConnectionAndPermissions(
-                this._xrayUrl,
-                this._username,
-                this._password,
-                this._accessToken,
-                prompt,
-                this._logManager
-            ))
+            !(await ConnectionUtils.checkXrayPermissions(this._xrayUrl, this._username, this._password, this._accessToken, prompt, this._logManager))
         ) {
-            return false;
+            this.deleteCredentialsFromMemory();
+            return TestConnectionStatus.FailedBadCredentials;
```

Comment:
How can you determine that this is a Bad Credential (401) error rather than:
* A permissions problem (403)
* An inactive server - I'm not sure if it is relevant anymore but worth checking
* Any other error, like 5xx

Take a look at testComponentPermission. I recommend returning the precise status from this function.